src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Middlewares/HttpResponseExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,60 @@
+using System.Net;
+using System.Reflection;
+
+namespace TodoTemplate.Api.Middlewares;
+
+public class HttpResponseExceptionHandlerMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public HttpResponseExceptionHandlerMiddleware(RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    public async Task Invoke(HttpContext context, IHostEnvironment webHostEnvironment)
+    {
+        context.Response.Headers.Add("Request-ID", context.TraceIdentifier);
+
+        try
+        {
+            await _next(context);
+        }
+        catch (Exception e)
+        {
+            var exception = UnWrapException(e);
+            var isKnownException = exception is KnownException;
+            var statusCode = (int)(exception is RestException restExp ? restExp.StatusCode : HttpStatusCode.InternalServerError);
+            
+            RestExceptionPayload restExceptionPayload = new RestExceptionPayload
+            {
+                Message = isKnownException || webHostEnvironment.IsDevelopment() ? exception.Message : nameof(UnknownException),
+                ExceptionType = isKnownException ? exception.GetType().FullName : typeof(UnknownException).FullName
+            };
+
+            if (exception is ResourceValidationException validationException)
+            {
+                restExceptionPayload.Details = validationException.Details;
+            }
```

Comment:
please extract a method to create the `restExceptionPayload` in order to make the `Invoke` method simpler and easier to read.